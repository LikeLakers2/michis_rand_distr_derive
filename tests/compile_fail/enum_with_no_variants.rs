use michis_rand_distr_derive::StandardDistribution;

#[derive(StandardDistribution)]
pub enum DoesNotSupportEnumWithNoVariants {}

fn main() {}
