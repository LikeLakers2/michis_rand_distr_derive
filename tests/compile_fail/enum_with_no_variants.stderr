error: proc-macro derive panicked
 --> tests/compile_fail/enum_with_no_variants.rs:3:10
  |
3 | #[derive(StandardDistribution)]
  |          ^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Cannot derive StandardDistribution for enums with zero variants
